### PR TITLE
Make sure 'no post text' exists before manipulating it

### PR DESCRIPTION
**Pull request by mleray**
### Description

Otherwise we get a console error that it is undefined:

Screenshot 2024-07-24 at 10 56 50

### Testing

On any page, on `main` branch you should see the above error, whereas with the changes on this branch it should be gone.

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [pandora](https://www-dev.greenpeace.org/test-pandora) | [admin](https://www-dev.greenpeace.org/test-pandora/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-pandora/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-pandora) | [composer-local.json](https://github.com/greenpeace/planet4-test-pandora/blob/main/composer-local.json)

:watch: 2024.07.24 09:17:11

File: assets/src/js/query-no-posts.js [hunk 1]
```diff
@@ -2,8 +2,10 @@ export const removeNoPostText = () => {
...
 
-  if (postTitle.innerHTML === '' && postDescription.innerHTML === '') {
-    const noResultsText = document.querySelector('.p4-query-loop .wp-block-query-no-results');
-    noResultsText.style.display = 'none';
+  if (!postTitle || !postDescription || postTitle.innerHTML !== '' || postDescription.innerHTML !== '') {
+    return;
   }
+
+  const noResultsText = document.querySelector('.p4-query-loop .wp-block-query-no-results');
+  noResultsText.style.display = 'none';
```
Comment (mardelnet) [inline assets/src/js/query-no-posts.js:10]: Should we check that `noResultsText` is not `undefined`?
```suggestion
  if(noResultsText) noResultsText.style.display = 'none';
```

File: assets/src/js/query-no-posts.js [hunk 2]
```diff
@@ -2,8 +2,10 @@ export const removeNoPostText = () => {
   const postTitle = document.querySelector('.p4-query-loop .wp-block-heading');
   const postDescription = document.querySelector('.p4-query-loop p');
 
-  if (postTitle.innerHTML === '' && postDescription.innerHTML === '') {
-    const noResultsText = document.querySelector('.p4-query-loop .wp-block-query-no-results');
-    noResultsText.style.display = 'none';
+  if (!postTitle || !postDescription || postTitle.innerHTML !== '' || postDescription.innerHTML !== '') {
```
Comment (mardelnet) [inline assets/src/js/query-no-posts.js:5]: Can't we simplify it a bit like this?
```suggestion
  if (postTitle?.innerHTML !== '' || postDescription?.innerHTML !== '') {
```

**Review by mardelnet — COMMENTED** [review]
Good job @mleray, I've left some minor suggestions.

File: assets/src/js/query-no-posts.js (hunk 2 above)
Comment (mleray) [reply assets/src/js/query-no-posts.js:5]: We absolutely can :D thanks!

File: assets/src/js/query-no-posts.js (hunk 1 above)
Comment (mleray) [reply assets/src/js/query-no-posts.js:10]: You have a good point 🤔 It should always be there in our Query block variations, but editors can make their own!

**Review by mardelnet — APPROVED** [review]
Well done 👍🏽

**Pull request merged**